presto-main/src/test/java/io/prestosql/operator/BenchmarkUnnestOperator.java

```diff
@@ -0,0 +1,424 @@
+            this.maxNestedCardinality = maxNestedCardinality;
+            this.maxStringLength = maxStringLength;
+        }
+
+        /**
+         * Generates a block with {@code entries} positions given an input {@code type}.
+         *
+         * {@code primitiveElementNulls} indicates what percentage of primitive elements
+         * (VARCHAR or INT) should be null. Similarly, the value {@code rowElementNulls}
+         * indicates the percentage for Row elements.
+         *
+         * The null percentages are not propagated inside a MapBlock's keyBlock. Everything
+         * is always non-null inside the returned keyBlock.
+         *
+         * The set S of valid input for {@code type} can be defined as:
+         * S = {VARCHAR, INTEGER, ROW(x), ARRAY(x), MAP(x, x)} where x belongs to S
+         *
+         */
+        public Block produceBlock(Type type, int entries, double primitiveElementNulls, double rowElementNulls)
+        {
+            if (type instanceof ArrayType) {
+                return produceArrayBlock((ArrayType) type, entries, primitiveElementNulls, rowElementNulls);
+            }
+            else if (type instanceof MapType) {
+                return produceMapBlock((MapType) type, entries, primitiveElementNulls, rowElementNulls);
+            }
+            else if (type instanceof RowType) {
+                return produceRowBlock((RowType) type, entries, primitiveElementNulls, rowElementNulls);
+            }
+            else if (type == VARCHAR) {
+                return produceStringBlock(entries, primitiveElementNulls);
+            }
+            else if (type == INTEGER) {
+                return produceIntBlock(entries, primitiveElementNulls);
+            }
+
+            throw new RuntimeException("not supported");
+        }
+
+        private Block produceStringBlock(int entries, double primitiveNulls)
+        {
+            BlockBuilder builder = VARCHAR.createBlockBuilder(null, 100);
+
+            for (int i = 0; i < entries; i++) {
+                if (random.nextDouble() <= primitiveNulls) {
+                    builder.appendNull();
+                }
+                else {
+                    VARCHAR.writeString(builder, generateRandomString(maxStringLength));
+                }
+            }
+
+            return builder.build();
+        }
+
+        private Block produceIntBlock(int entries, double primitiveNulls)
+        {
+            BlockBuilder builder = INTEGER.createBlockBuilder(null, 100);
+
+            for (int i = 0; i < entries; i++) {
+                if (random.nextDouble() < primitiveNulls) {
+                    builder.appendNull();
+                }
+                else {
+                    INTEGER.writeLong(builder, random.nextInt(INT_UPPER_LIMIT));
+                }
+            }
+
+            return builder.build();
+        }
+
+        private Block produceArrayBlock(ArrayType arrType, int entries, double primitiveNulls, double rowElementNulls)
```

Comment:
Don't abbreviate variable names (`arrType`). Also rename the other variables as per the earlier comments.